### PR TITLE
Deploy forbidden page

### PR DESCRIPTION
**Pull request by andrewkrug**
This deploys the new forbidden page needed by the access proxy and others.

### CODE REVIEW

**Comment by andrewkrug** [comment]
Created CHG0012137

File: dashboard/app.py [hunk 1]
```diff
@@ -159,6 +159,18 @@ def page_not_found(error):
     return render_template('404.html'), 404
 
 
+@app.route('/forbidden')
+def forbidden():
+    """Route to render error page."""
+    reason = None
+    if 'reason' in request.args:
+        reason = request.args['reason']
+        logger.error(
```
Comment (akatsoulas) [inline dashboard/app.py]: nit: You could simplify this block:

```python
reason = request.args.get('reason')
if reason:
    logger.error(......)
```

File: dashboard/models/user.py [hunk 2]
```diff
@@ -24,59 +24,75 @@ def api_url(self):
...
-                if mozillians_response.status_code is not 200:
-                    return None
-                response = mozillians_response.json()
-                return response
-            except (requests.exceptions.Timeout, requests.exceptions.ConnectionError):
-                return None
-        else:
+    def _get_user_info(self, email):
+        response = self._get_user(email)
+        if not response:
```
Comment (akatsoulas) [inline dashboard/models/user.py]: nit: `if not self._get_user(email)`

File: dashboard/models/user.py [hunk 3]
```diff
@@ -24,59 +24,75 @@ def api_url(self):
...
             return None
 
-    def _is_only_one_avatar(self, response):
-        # Check if only single resource gets returned and it's valid
-        avatars = response.get('results', -1)
+        try:
+            results = response.get('results', -1)
+            user_url = results[0].get('_url')
+            user_info = requests.get(user_url, headers=self.headers, timeout=5)
```
Comment (akatsoulas) [inline dashboard/models/user.py]: I would suggest to have in the try block only the line that might throw an exception.

File: dashboard/models/user.py [hunk 4]
```diff
@@ -24,59 +24,75 @@ def api_url(self):
...
+        except:
+            return None
 
-        if len(avatars) == 1:
-            self.user_url = avatars[0].get('_url')
-            return True
-        else:
-            self.user_url = None
-            return False
+        if user_info.status_code is not 200:
```
Comment (akatsoulas) [inline dashboard/models/user.py]: s/is not/!=

File: dashboard/models/user.py [hunk 5]
```diff
@@ -24,59 +24,75 @@ def api_url(self):
...
 
-    def _get_image_url(self):
-        # Finally fetch user public avatar and make sure  we have a valid fallback
         try:
-            response = requests.get(self.user_url, headers=self.headers, timeout=5).json()
-            if response['photo']['privacy'] == 'Public':
-                return response['photo']['value']
-        except (requests.exceptions.Timeout, requests.exceptions.ConnectionError):
+            info = user_info.json()
+        except:
```
Comment (akatsoulas) [inline dashboard/models/user.py]: Is there a specific exception to catch?

File: dashboard/models/user.py [hunk 6]
```diff
@@ -24,59 +24,75 @@ def api_url(self):
...
-        else:
-            avatar_url = None
+        try:
+            mozillians_response = requests.get(self.api_url, headers=self.headers,
+                                               params=params, timeout=5)
+        except:
+            return None
 
-        return avatar_url
+        if mozillians_response.status_code is not 200:
```
Comment (akatsoulas) [inline dashboard/models/user.py]: s/is not/!=

File: dashboard/models/user.py [hunk 7]
```diff
@@ -24,59 +24,75 @@ def api_url(self):
...
+            return None
 
-        if response and self._is_only_one_avatar(response):
-            avatar_url = self._get_image_url()
-        else:
-            avatar_url = None
+        try:
+            mozillians_response = requests.get(self.api_url, headers=self.headers,
+                                               params=params, timeout=5)
+        except:
```
Comment (akatsoulas) [inline dashboard/models/user.py]: Same here.

**Pull request merged**